### PR TITLE
[bug][typescript] Fix node client generator import file paths

### PR DESCRIPTION
**Pull request by mahirk**
### Bug and Recap
Issue: Resolves https://github.com/OpenAPITools/openapi-generator/issues/7385
> For mapping such as:
> ```
>   "importMappings": {
>     "Pet": "../dataModel/petModel"
>   }
> ```
> **Actual**
> ```
> import { AModelPetModel } from '../../dataModel/petModel';
> ```
> **Expected**
> ```
> import { Pet } from '../dataModel/petModel';
> ```
> 
> For a mapping such as:
> ```
>   "importMappings": {
>     "Pet": "@company/prefix-zoo-store"
>   }
> ```
> 
> **Actual**
> ```
> import { NyPrefixZooStore } from '../@company/prefix-zoo-store';
> ```
> **Expected**
> ```
> import { Pet } from '@company/prefix-zoo-store';
> ```

### Method
Below is the map in `operations.get("imports")` based on the import location:

_Model is created by generator_
```
[{import=model/pet, classname=Pet}]
```

_BYOModel_
```
[{import=@company/prefix-zoo-store, classname=Pet}]
```

Therefore the logic was updated to a simpler version, assuming the mapping is created by prior areas.

```
for (Map<String, Object> im : imports) {
     im.put("filename", im.get("import").toString());
}
```

This resulted in the following import for:

_Created Model_
```
import { Pet } from '../model/pet';
```

_BYOModel_
```
import { Pet } from '../@company/prefix-zoo-store';
```

Per the above, this fixed the issue of the improper render which led to the bad `classname` i.e. what led to the following `NyPrefixZooStore`

The next issue was the extra `../` which is embedded as part of the template. See: https://github.com/OpenAPITools/openapi-generator/blob/master/modules/openapi-generator/src/main/resources/typescript-node/api-single.mustache#L8

For this purpose, the extra directory listing was remove from the template and added into the `Codegen` itself, where the `../` is appended by default _ONLY_ when the model is derived from an auto generated file. 

After this change here are the mappings in `operations.get("imports")`

_Model is created by generator_
```
[{import=../model/pet, classname=Pet}]
```

_BYOModel_
```
[{import=@company/prefix-zoo-store, classname=Pet}]
```

This resulted in the following import for:

_Created Model_
```
import { Pet } from '../model/pet';
```

_BYOModel_
```
import { Pet } from '@company/prefix-zoo-store';
```

#### Unit
1. Updated tests for `defaultModelImportTest` 
2. Added tests for `postProcessOperationsWithModels`

#### Manual
1. Ran the above expected v/s actual output comparisons to ensure the values provided are accurate

### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] If contributing template-only or documentation-only changes which will change sample output, [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) beforehand.
- [x] Run the shell script `./bin/generate-samples.sh`to update all Petstore samples related to your fix. This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. These must match the expectations made by your contribution. You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master`
- [x] Copy the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) to review the pull request if your PR is targeting a particular programming language.

### CODE REVIEW

**Comment by mahirk** [comment]
Tagging the most recent contributors to the typescript generator per the contributor list:
@petejohansonxo 
@amakhrov
cc: @wing328

**Comment by mahirk** [comment]
Tagging in a few more TS contributors from [the technical commitee](https://github.com/OpenAPITools/openapi-generator#62---openapi-generator-technical-committee) to help have a look at this:
@Vrolijkx @macjohnny @topce

**Comment by mahirk** [comment]
+ @jimschubert as another active maintainer for a review 
+ @sebastianhaas as a TS Contributor

File: modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/TypeScriptNodeClientCodegen.java
Comment (amakhrov) [inline modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/TypeScriptNodeClientCodegen.java:226]: looks like now the `getModelnameFromModelFilename` method is no longer used and can be removed?

File: modules/openapi-generator/src/test/java/org/openapitools/codegen/typescript/typescriptnode/TypeScriptNodeClientCodegenTest.java
Comment (amakhrov) [inline modules/openapi-generator/src/test/java/org/openapitools/codegen/typescript/typescriptnode/TypeScriptNodeClientCodegenTest.java:160]: Perhaps just nitpicking - but maybe there could be a better name for this method? Right now it's kind of unclear what its intention is (unless I read the actual method body).

**Review by amakhrov — COMMENTED** [review]
Thanks for your contribution! The fix looks reasonable to me (just some minor suggestions).

I'm also wondering if this issue (and the fix) are typescript-node specific, or other typescript generators suffer from the same bug.

**Comment by mahirk** [comment]
Thanks for the review @amakhrov! Made the suggestions as recommended above. 
The bug is likely other ts generators. Though it does mean more reviews, once this is approved, I was hoping to break down a pr for the current ts generators (and additionally try and add a sample as well so that it can be well tested)
Here is my testing so far:

| Generator | Issue                                                                                                                                                                                                                              | LoE                                                            |
|-----------|------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|----------------------------------------------------------------|
| node      | does not use import mappings correctly                                                                                                                                                                                             | low                                                            |
| axios     | does not support import mappings                                                                                                                                                                                                   | medium                                                         |
| fetch     | does not use import mappings correctly + expects "to" and "from" json methods                                                                                                                                                      | medium to high (will need to think of a better way to do this) |
| angular   | the template assumes everything to be in the mode directory, hence it fails https://github.com/OpenAPITools/openapi-generator/blob/master/modules/openapi-generator/src/main/resources/typescript-angular/api.service.mustache#L11 | low                                                            |


I'll potentially look at a quick fix for angular next and then attempt to evaluate axios and fetch since those are popular libraries. 

Another issue for typescript node is: https://github.com/OpenAPITools/openapi-generator/issues/7440

**Review by macjohnny — APPROVED** [review]
LGTM

File: modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/TypeScriptNodeClientCodegen.java
Comment (jimschubert) [inline modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/TypeScriptNodeClientCodegen.java:166]: @mahirk not a blocker on this pr, but wanted to comment since you said you were planning to look at ts implementation in other generators...

since modelPackage is passed by the user, this could result in `models//` when passed with trailing slash. It might be best to use StringUtils, something like:

```
return DEFAULT_MODEL_IMPORT_DIRECTORY_PREFIX + 
    StringUtils.appendIfMissing(modelPackage(), "/") +
    camelize(toModelName(name), true);
```

**Review by jimschubert — APPROVED** [review]
looks good. Just left a small comment if you're looking to do the same in other generators.

**Comment by tomh4** [comment]
was this ever 

> Thanks for the review @amakhrov! Made the suggestions as recommended above.
> The bug is likely other ts generators. Though it does mean more reviews, once this is approved, I was hoping to break down a pr for the current ts generators (and additionally try and add a sample as well so that it can be well tested)
> Here is my testing so far:
> 
> Generator	Issue	LoE
> node	does not use import mappings correctly	low
> axios	does not support import mappings	medium
> fetch	does not use import mappings correctly + expects "to" and "from" json methods	medium to high (will need to think of a better way to do this)
> angular	the template assumes everything to be in the mode directory, hence it fails https://github.com/OpenAPITools/openapi-generator/blob/master/modules/openapi-generator/src/main/resources/typescript-angular/api.service.mustache#L11	low
> I'll potentially look at a quick fix for angular next and then attempt to evaluate axios and fetch since those are popular libraries.
> 
> Another issue for typescript node is: #7440

Did you ever figure out how to get this working with typescript-fetch ? The importMappings still seem to be ignored completely

**Pull request merged**
